kobo/hub/models.py

```diff
@@ -260,7 +304,49 @@ def _get_relative_log_path(self, name):
             raise RuntimeError("Invalid log normpath.")
         return log_path
 
+    def get_chunk(self, name, offset=0, length=-1):
+        """Returns a sequence of bytes from the named log.
+
+        offset -- optional start index of read
+        length -- optional length; if omitted, read to end of file
+
+        Will try to avoid returning a chunk ending in the middle of
+        a single UTF-8 character, therefore the returned length may
+        be less than requested."""
+
+        name = self._get_relative_log_path(name)
+
+        if name in self.cache:
+            # already loaded so no point in reading again
+            end = None if length < 0 else offset + length
+            return self.cache[name][offset:end]
+
+        # not loaded; read just this part
+        log_path = self._get_absolute_log_path(name)
+        log_file = None
+        try:
+            if os.path.isfile(log_path):
+                log_file = open(log_path, 'rb')
```

Comment:
No I didn't.

Looks like it would require changing the API semantics such that the offset parameter is no longer a byte offset but a character offset.  Then you can never request to start reading in the middle of a character.

I'm not 100% sure if that's going to be practical but if it works then seems worth doing.  Thanks, I'll look into it.